autofs-5.1.8 - fix missing unlock in sasl_do_kinit_ext_cc()

From: James Dingwall <james-autofs@dingwall.me.uk>

There is a missing mutex unlock in function sasl_do_kinit_ext_cc(),
fix it.

Signed-off-by: James Dingwall <james-autofs@dingwall.me.uk>
Signed-off-by: Ian Kent <raven@themaw.net>
---
 CHANGELOG            |    1 +
 modules/cyrus-sasl.c |    4 ++++
 2 files changed, 5 insertions(+)

--- autofs-5.1.8.orig/CHANGELOG
+++ autofs-5.1.8/CHANGELOG
@@ -17,6 +17,7 @@
 - remove nonstrict parameter from tree_mapent_umount_offsets().
 - fix handling of incorrect return from umount_ent().
 - dont use initgroups() at spawn.
+- fix missing unlock in sasl_do_kinit_ext_cc().
 
 19/10/2021 autofs-5.1.8
 - add xdr_exports().
--- autofs-5.1.8.orig/modules/cyrus-sasl.c
+++ autofs-5.1.8/modules/cyrus-sasl.c
@@ -721,6 +721,10 @@ sasl_do_kinit_ext_cc(unsigned logopt, st
 
 	debug(logopt, "Kerberos authentication was successful!");
 
+	status = pthread_mutex_unlock(&krb5cc_mutex);
+	if (status)
+		fatal(status);
+
 	return 0;
 
 out_cleanup_def_princ:
